### ISSUE TITLE
Datetimeoffset to int convertation not allowed mssql

### ISSUE THREAD

**Issue by BlackGad** [HeidiSQL/HeidiSQL#98]
## Expected behavior

Correctly show table data

## Current behavior

When i click data tab popup window appears
![image](https://user-images.githubusercontent.com/8100521/35217758-0ba3bf88-ff75-11e7-8b73-adef8d4f4777.png)

## Failure information (for bugs)

Table columns in VS SQL provider
![image](https://user-images.githubusercontent.com/8100521/35217859-63a1f510-ff75-11e7-982c-f4c6c8f7c593.png)
Table columns in HeidiSQL
![image](https://user-images.githubusercontent.com/8100521/35228120-9b5f9110-ff98-11e7-9a66-f378f6bfe989.png)
On HeidiSQL data expand
![image](https://user-images.githubusercontent.com/8100521/35217913-8686e9b4-ff75-11e7-9f66-dc2db4d624a3.png)

### Steps to reproduce

1) Create DB in MSSQL with table which contains datetimeoffset column
2) View it with HeidiSQL

### Context

* HeidiSQL version:
![image](https://user-images.githubusercontent.com/8100521/35217986-bc814460-ff75-11e7-80aa-430764a132e1.png)
* Database system + version: SQL Express 2016
* Operating system: WIN 10 x64

**Comment by ansgarbecker** [comment]
Please provide the SQL query which causes the error, from the bottom log panel.

**Comment by BlackGad** [comment]
```sql
SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA='dbo' AND TABLE_NAME='OrganizationInviteTickets';
SELECT c.name AS "column", prop.value AS "comment" FROM sys.extended_properties AS prop INNER JOIN sys.all_objects o ON prop.major_id = o.object_id INNER JOIN sys.schemas s ON o.schema_id = s.schema_id INNER JOIN sys.columns AS c ON prop.major_id = c.object_id AND prop.minor_id = c.column_id WHERE   prop.name='MS_Description'  AND s.name='dbo'  AND o.name='OrganizationInviteTickets';
SELECT C.CONSTRAINT_NAME, C.CONSTRAINT_TYPE, K.COLUMN_NAME FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS AS C INNER JOIN INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS K ON   C.CONSTRAINT_NAME = K.CONSTRAINT_NAME   AND K.TABLE_NAME='OrganizationInviteTickets'   AND K.TABLE_SCHEMA='dbo' WHERE C.CONSTRAINT_TYPE IN ('PRIMARY KEY', 'UNIQUE') ORDER BY K.ORDINAL_POSITION;
SELECT TOP 1000  "Id",  "CreatorUserID",  "GuestUserID",  "OrganizationID",  "Created",  "Modified",  "Disabled",  CAST("Timestamp" AS INT),  CAST("Expiry" AS INT) FROM "CinegyLicensePortal"."dbo"."OrganizationInviteTickets";
/* SQL Error (529): Explicit conversion from data type datetimeoffset to int is not allowed. */
```
It seems that error here that HeidiSQL determine wrong column types

**Comment by BlackGad** [comment]
ahh, sorry attached wrong image "Table columns in HeidiSQL" updated above

**Comment by BlackGad** [comment]
Seems in your cycle datetimeoffset is unexpected and column type become from prev column

**Comment by ansgarbecker** [comment]
Some years ago, in c5192252b05671bcac503b0d1039ddcbca6ec5a7 I fixed the wrong casting from TIMESTAMP columns from a binary value. Now on SQL server 2017, I have still no error with that CAST to INT. On SQL Server 2016, as you showed, you get an error, in even older versions there were no error.

I also found this old forum thread: https://www.heidisql.com/forum.php?t=13041

I suppose HeidiSQL just has to support (= detect) this DATETIMEOFFSET data type, and sort it into the temporal category most probably.
